### PR TITLE
Skip redundant schedules

### PR DESCRIPTION
**Pull request by johnwatson484**

### CODE REVIEW

File: app/processing/schedule/schedule-pending-payment-schedules.js [hunk 1]
```diff
@@ -11,8 +12,9 @@ const schedulePendingPaymentSchedules = async () => {
     const scheduledPaymentSchedules = await getScheduledPaymentSchedules(started, transaction)
     const validScheduledPaymentSchedules = getValidScheduled(scheduledPaymentSchedules)
     const updatedScheduledPaymentSchedules = await getUpdatedSchedule(validScheduledPaymentSchedules, started, transaction)
+    const removedDefunctPaymentSchedules = await getRemovedDefunctPaymentSchedules(updatedScheduledPaymentSchedules, started, transaction)
```
Comment (simonnineteen90) [inline app/processing/schedule/schedule-pending-payment-schedules.js:15]: is this returning the removed defucted payment schedules or is it returning the payment schedules with the defunct ones removed?

File: app/processing/payment-request/get-completed-payment-request-by-correlation-id.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+const db = require('../../data')
```
[the remark this answers is not in the document]
Comment (johnwatson484) [reply app/processing/payment-request/get-completed-payment-request-by-correlation-id.js:1]: We have `paymentRequestId` of the In Progress payment request.  We need to find a matching completed one/s.  Simplest was is the `correlationId` which ties each state together.

File: app/processing/payment-request/get-previous-payment-requests-with-payment-schedules.js.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+const { getCompletedSchedule } = require('../schedule')
+const getCompletedPaymentRequestByCorrelationId = require('./get-completed-payment-request-by-correlation-id')
+
+const getPreviousPaymentRequestsWithPaymentSchedules = async (previousPaymentRequests, transaction) => {
+  const previousPaymentRequestsWithSchedules = []
+
+  for (const paymentRequest of previousPaymentRequests) {
+    // Always allow first payment request so we always have a payment request to compare against
+    if (paymentRequest.paymentRequestNumber === 1) {
```
[the remark this answers is not in the document]
Comment (johnwatson484) [reply app/processing/payment-request/get-previous-payment-requests-with-payment-schedules.js.js:9]: PR1 will never have a schedule as we only generate schedules for post payment adjustments.  But we always need a baseline to compare to, which is PR1.

File: app/processing/schedule/schedule-pending-payment-schedules.js (hunk 1 above)
Comment (johnwatson484) [reply app/processing/schedule/schedule-pending-payment-schedules.js:15]: Returning payment schedules with defunct ones removed

**Review by simonnineteen90 — APPROVED** [review]

**Pull request merged**